portfolio/src/main/webapp/script.js

```diff
@@ -100,22 +99,13 @@ function UserLogin(){
     });
 }
 
-    function getURL(){
-        fetch('/user?loginUrl').then(loginUrl => loginUrl.text()).then(loginUrl => String(loginUrl)).then(loginUrl => loginUrl.trim()).then((loginUrl) => {
-          document.getElementById("notLoggedIn").innerHTML = loginUrl;
-        });
-        fetch('/user?logoutUrl').then(logoutUrl => logoutUrl.text()).then(logoutUrl => String(logoutUrl)).then(logoutUrl => logoutUrl.trim()).then((logoutUrl) => {
-            document.getElementById("logOut").innerHTML = logoutUrl;
-    });
- }
+function getEmail(){
```

Comment:
This function currently errors when it is passed Logs even though logs is the list coming from the server?

Reply:
Not sure about this one, but it could be because you're returning an ArrayList right now.  Try removing the ArrayList as per my comments above, then see if this still gives you problems.